### PR TITLE
executor: create vecgroupcheck package to improve test

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/44940
Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/internal/vecgroupchecker/vecgroupchecker.go [hunk 1]
```diff
@@ -0,0 +1,514 @@
+// Copyright 2023 PingCAP, Inc.
```
Comment (Defined2014) [inline executor/internal/vecgroupchecker/vecgroupchecker.go:1]: I think `vec_group_checker.go` is better.

File: executor/internal/vecgroupchecker/vecgroupchecker_test.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
+// Copyright 2023 PingCAP, Inc.
```
Comment (Defined2014) [inline executor/internal/vecgroupchecker/vecgroupchecker_test.go:1]: ditto, rename to `vec_group_checker_test.go` is better.

Comment (hawkingrei) [reply executor/internal/vecgroupchecker/vecgroupchecker_test.go]: updated

File: executor/internal/vecgroupchecker/vecgroupchecker.go (hunk 1 above)
Comment (hawkingrei) [reply executor/internal/vecgroupchecker/vecgroupchecker.go]: updated

**Review by Defined2014 — APPROVED** [review]

**Comment by hawkingrei** [comment]
@XuHuaiyu @wshwsh12 PTAL

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**